### ISSUE TITLE
Dark mode

### ISSUE THREAD

**Issue by ayberkt** [ncatlab/nlab#10]
It would be quite convenient to be able to read nlab in dark mode. As far as I know, it is not too hard to achieve this by minor tweaks to the CSS.

Is this something that should be addressed directly by [instiki](https://github.com/parasew/instiki) maybe? (CC: @parasew)

**Comment by Dorson** [comment]
Made exactly that for the night and evening time.

See live demo on GitHub page:

https://dorson.github.io/CSS-Dark-Mode-and-color-switch/

Have fun, feel free to fork, copy, adopt, port, etc...

**Comment by BinaryQuantumSoul** [comment]
Here are some community-made dark themes:

https://nforum.ncatlab.org/discussion/17886/
https://nforum.ncatlab.org/discussion/18734/

**Comment by ccatrett** [comment]
I would like to work on trying to integrate one of these community made dark themes into the website itself instead of as an external download.

**Comment by BinaryQuantumSoul** [comment]
In my dark theme, there is a significative part that is about mobile friendliness and modernity regardless of the theme brightness. I'd also like to integrate them into the base website. The dark part could be a toggle

**Comment by ccatrett** [comment]
@BinaryQuantumSoul would you want to work on it together? I am a relatively new to contributing to open source projects but I think this could be a good feature to work on.

**Comment by BinaryQuantumSoul** [comment]
Of course, though we must first make sure the project maintainer @sattlerc is interested, so we don't push work for nothing. To be clear, the three things I'd like to do is:
1. Make the website mobile friendly
2. Modernize the light theme
3. Add a dark theme toggle

Can you try my dark theme (it's the second I linked) and give your opinion @ccatrett ? Also try to resize it to see the mobile part.

**Comment by sattlerc** [comment]
We are happy to merge any work on the user interface as long as:
* the users like it (we can create a test instance if you have problems setting one up yourself),
* the code changes don't make the code base more complicated to maintain.

**Comment by ccatrett** [comment]
@BinaryQuantumSoul I tried your dark theme and I really like the mobile support along with the overall color scheme and styles.
Here are some ways that I believe it could be improved:

1.  The text on the ``All Pages`` is very dark and could be improved for readability.  Here is a comparison of the colors between another page with links and the ``All Pages`` page:

![Image](https://github.com/user-attachments/assets/c6ebdc4f-ba3e-4fba-8bf2-c2c82bb2e1d4)

![Image](https://github.com/user-attachments/assets/02ab0922-4f38-4381-a157-f20ad00ed178)

2. As mentioned before, I am a fan of the mobile support. However, I believe that on wider screens, it stretches too far and becomes a bit difficult to read. Another comparison:

![Image](https://github.com/user-attachments/assets/1d0d90a4-b9c9-43c5-9730-847320c3e005)

![Image](https://github.com/user-attachments/assets/bb325e03-c45c-4a8d-909d-82f0628ca611)

![Image](https://github.com/user-attachments/assets/528d01a4-1cbb-4447-9e13-5993e9ff6ed3)

I think that either of the latter widths are easier to read since your eyes don't have to scan the entire length of the viewing window. Potential solutions for this could be limiting the text width or adding a toggle feature to change the desired text width, like Wikipedia has recently done.

3. The bright green surrounding the search bar was a bit jarring when I first saw it. I think the intensity of the green here could be decreased or potentially changed to a different color (although, I like the color as is):

![Image](https://github.com/user-attachments/assets/f8660098-721e-4abb-aa85-58608ca26421)

These were my first impressions when I tried your dark theme. Again, overall, I enjoy this dark theme and the above are the areas where I think it could be improved.

With regards to your points 1., 2., and 3., I am in agreeance; I think that these additions to the website would make it more visually appealing and modern.

**Comment by BinaryQuantumSoul** [comment]
> We are happy to merge any work on the user interface as long as:
> * the users like it (we can create a test instance if you have problems setting one up yourself),
> * the code changes don't make the codebase more complicated to maintain.

Perfect !

**Comment by BinaryQuantumSoul** [comment]
> I tried your dark theme and I really like the mobile support along with the overall color scheme and styles.

Glad you liked it!

> 1.  The text on the ``All Pages`` is very dark and could be improved for readability. 

You are right, that one and also the purple color for some links look a bit off. For example look at search results page.

> 2. As mentioned before, I am a fan of the mobile support. However, I believe that on wider screens, it stretches too far and becomes a bit difficult to read.

Are you talking about wide mobile screen or wide monitors overall ? For the first case we can do a specific media query for that middle size. For the big screen I had the feeling like the prior was a waste of space, but you are right we could also add a small toggle for the width.

> 3. The bright green surrounding the search bar was a bit jarring when I first saw it. I think the intensity of the green here could be decreased or potentially changed to a different color (although, I like the color as is):

In my opinion this whole search box is a bit odd since it's huge, and it goes to google search... but the color is indeed too bright, it's also used in some pages where people add in-article debates, we should find another style

> With regards to your points 1., 2., and 3., I am in agreeance; I think that these additions to the website would make it more visually appealing and modern.

Perfect

**Comment by ccatrett** [comment]
>Are you talking about wide mobile screen or wide monitors overall ? For the first case we can do a specific media query for that middle size. For the big screen I had the feeling like the prior was a waste of space, but you are right we could also add a small toggle for the width.

I was referring to wide monitors overall, sorry about the confusion. I think the media query sounds like a great idea!

What do you think some next steps would be?

**Comment by ccatrett** [comment]
@BinaryQuantumSoul Have you been able to test the website locally yet?

**Comment by BinaryQuantumSoul** [comment]
Nope tested a whole day without having it working but now I won't have time until july
